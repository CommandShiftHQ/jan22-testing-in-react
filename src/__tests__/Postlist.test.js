import { fireEvent, render, screen } from "@testing-library/react";
import PostList from "../components/Postlist";
import renderer from "react-test-renderer";

describe("PostList", () => {
  const validProps = {
    posts: [
      {
        id: 1,
        title: "test title",
        body: "test body",
        tags: ["tag 1", "tag 2", "tag 3"],
        author: "test author",
        date: "test date",
        isPublished: true,
      },
      {
        id: 2,
        title: "test title2",
        body: "test body2",
        tags: ["tag 1", "tag 2", "tag 3"],
        author: "test author2",
        date: "test date2",
        isPublished: false,
      },
    ],
  };
  test("check snapshot matches", () => {
    const rendered = renderer.create(<PostList posts={validProps.posts} />);

    expect(rendered).toMatchSnapshot();
  });

  test("Assert last upvoted post is displayed", () => {
    render(<PostList posts={validProps.posts} />);

    const buttons = screen.getAllByRole("button");
    expect(buttons).toHaveLength(2);

    const lastUpvoted = screen.queryByText("Last upvoted post");
    expect(lastUpvoted).toBeNull();

    fireEvent.click(buttons[0]);
    expect(
      screen.getByText("Last upvoted post: test title")
    ).toBeInTheDocument();
  });
});
