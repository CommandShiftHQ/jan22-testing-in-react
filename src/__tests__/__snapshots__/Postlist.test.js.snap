// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList check snapshot matches 1`] = `
<div
  className="postlist"
>
  <div
    className="post"
  >
    <div
      className="post-heading"
    >
      <h2>
        test title
      </h2>
      <p>
        test body
      </p>
    </div>
    <div
      className="post-counter"
    >
      <span>
        Upvotes: 
        0
      </span>
      <button
        onClick={[Function]}
        type="button"
        value="test title"
      >
        Upvote this
      </button>
    </div>
    <div
      className="post-author"
    >
      Author: 
      test author
    </div>
    <div
      className="post-date"
    >
      Published: 
      test date
    </div>
    <h3>
      Tags:
    </h3>
    <ul>
      <li>
        tag 1
      </li>
      <li>
        tag 2
      </li>
      <li>
        tag 3
      </li>
    </ul>
  </div>
  <div
    className="post"
  >
    <div
      className="post-heading"
    >
      <h2>
        test title2
      </h2>
      <p>
        Coming soon!
      </p>
    </div>
    <div
      className="post-counter"
    >
      <span>
        Upvotes: 
        0
      </span>
      <button
        onClick={[Function]}
        type="button"
        value="test title2"
      >
        Upvote this
      </button>
    </div>
    <div
      className="post-author"
    >
      Author: 
      test author2
    </div>
    <div
      className="post-date"
    >
      Published: 
      test date2
    </div>
    <h3>
      Tags:
    </h3>
    <ul>
      <li>
        tag 1
      </li>
      <li>
        tag 2
      </li>
      <li>
        tag 3
      </li>
    </ul>
  </div>
</div>
`;
